Framework/src/TObject2JsonWorker.cxx

```diff
@@ -0,0 +1,152 @@
+      }
+
+      std::string payload = socketReceive();
+      if (payload.empty()) {
+        // empty or timeout
+        continue;
+      }
+
+      std::string response = handleRequest(payload);
+
+      // always success to send because of inproc communication
+      socketSend(identity, response);
+    }
+  }
+  catch (std::exception &e) {
+    // Context was terminated
+    QcInfoLogger::GetInstance() << "Closing worker" << infologger::endm;
+  }
+}
+
+std::string TObject2JsonWorker::handleRequest(std::string request)
+{
+  std::string agentName;
+  std::string objectName;
+  const auto slashIndex = request.find_first_of('/');
+
+  if (std::string::npos == slashIndex) {
+    QcInfoLogger::GetInstance() << "Wrong request request received: '" << request << "'" << infologger::endm;
+    return std::string("{\"request\": \"") + request + std::string("\", ") +
+      std::string("\"error\": 400, \"message\": \"Wrong request\", ") +
+      std::string("\"why\": \"Ill-formed path, slash required\"}");
+  }
+
+  agentName = request.substr(0, slashIndex);
+  objectName = request.substr(slashIndex + 1);
+
+  try {
+    // calls to getJsonObject implies the process to have a 'Interrupted system call', why?
+    std::string result = mBackend->getJsonObject(agentName, objectName);
+    if (result.empty()) {
+      QcInfoLogger::GetInstance() << "Object not found '" << request << "'" << infologger::endm;
+      return std::string("{\"request\": \"") + request + std::string("\", ") +
+        std::string("\"error\": 404, \"message\": \"Not found\", ") +
+        std::string("\"why\": \"The requested object was not found\"}");
+    }
+
+    QcInfoLogger::GetInstance() << "Successful request: '" << request << "'" << infologger::endm;
+    return std::string("{\"request\": \"") +
+      request +
+      std::string("\", \"payload\": ") +
+      result +
+      std::string("}");
+  } catch (const std::exception& error) {
+    QcInfoLogger::GetInstance() <<
+      "Internal error for request '" << request << "': '" << error.what() << "'" <<
+      infologger::endm;
+    return std::string("{\"request\": \"") + request + std::string("\", ") +
+      std::string("\"error\": 500, \"message\": \"Internal error\", ") +
+      std::string("\"why\": \"") + error.what() + std::string("\"}");
+  }
+}
+
+//  Receive 0MQ string from socket and convert into string
+std::string TObject2JsonWorker::socketReceive()
+{
+  zmq::message_t message;
+  bool received = mWorkerSocket.recv(&message);
+  if (!received) {
+    // timeout
+    return std::string();
+  }
+  return std::string(static_cast<char*>(message.data()), message.size());
+}
+
+//  Sends string as 0MQ string, as multipart non-terminal
+bool TObject2JsonWorker::socketSend(const std::string & identity, const std::string & payload)
+{
+  zmq::message_t identityMessage(identity.size());
+  memcpy(identityMessage.data(), identity.data(), identity.size());
+
+  mWorkerSocket.send(identityMessage, ZMQ_SNDMORE);
```

Comment:
Why do you need `ZMQ_SNDMORE`? This specified that message is multipart.

Reply:
yes, we send two messages: one identifier and one payload